cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/messages/GroupedMessageIteratorProvider.java

```diff
@@ -133,11 +129,29 @@ public CompletableFuture<Iterator<StoredGroupedMessageBatch>> nextIterator()
 				{
 					PageId pageId = new PageId(book.getId(), cassandraFilter.getPage());
 					// Updated limit should be smaller, since we already got entities from previous batch
-					cassandraFilter = createNextFilter(cassandraFilter, Math.max(limit - returned.get(),0));
+					cassandraFilter = createNextFilter(cassandraFilter, Math.max(limit - returned.get(), 0));
 					return new ConvertingPagedIterator<>(resultSet, selectQueryExecutor, 0, new AtomicInteger(),
 							entity -> MessagesWorker.mapGroupedMessageBatchEntity(pageId, entity), converter::getEntity,
 							"fetch next page of message batches");
 				}, composingService)
 				.thenApplyAsync(it -> new FilteredGroupedMessageBatchIterator(it, filter, limit, returned), composingService);
 	}
+
+	private boolean isDirectOrder() {
+		return order == Order.DIRECT;
```

Comment:
this.firstPage = FilterUtils.findFirstPage(filter.getPageId(), filter.getFrom(), book);
		this.lastPage = FilterUtils.findLastPage(filter.getPageId(), filter.getTo(), book);

looks like we can get the same behavior if set
* from <= to in DIRECT order 
* from >= to in REVERSE order 
in other case we should check that from <= to in constructor